R/Terminator.R

```diff
@@ -59,6 +58,26 @@ Terminator = R6Class("Terminator",
     print = function(...) {
       catf(self$format())
       catf(str_indent("* Parameters:", as_short_string(self$param_set$values)))
+    },
+
+    #' @description
+    #' Subclasses overwrite this method to define the number of steps in the
+    #' progressor.
+    #'
+    #' @param archive ([Archive]).
+    #' @return `integer(1)`
+    progressr_steps = function(archive) {
+      stop("Abstract class")
+    },
+
+    #' @description
```

Comment:
The descriptions here are somewhat less helpful than the descriptions in the specific classes.

Reply:
Yes I indented to write a description for other developers since the description is overwritten in the subclasses. But this might be confusing, since terminators that do not support progressbars inherit this abstract method with the description.

Maybe we should remove the abstract methods in `Terminators`. Then we do not have a complete API description in `Terminators` but it is less confusing for the user. We can write under technical details that  `progressr_steps` and `progressr_update` can be implemented and are called by `eval_batch`

Reply:
Actually, if not every Terminator implements these progressr mehtods it should not be abstract, right? The design should be that all child classes have to implement the abstract methods. So what does that mean? Either we have a TerminatorProgressr class that inherits from Terminator or you just delete the abstract methods. At the moment I would opt for the latter. Also you could use templates again to avoid copy paste?

Reply:
Yes you are right. This should not have been an abstract class in the first place. I think a TerminatorProgressr class is little bit overkill. I opt for option 2.